pulsar-client-cpp/lib/ConsumerImpl.cc

```diff
@@ -879,4 +883,11 @@ void ConsumerImpl::brokerConsumerStatsListener(Result res, BrokerConsumerStatsIm
     }
 }
 
+bool ConsumerImpl::isReadCompacted() {
+    Lock lock(mutex_);
+    bool ret = readCompacted_;
+    lock.unlock();
+    return ret;
```

Comment:
This could be shorthened into:

```cpp
Lock lock(mutex_);
return readCompacted_;
```
Because the lock is automatically released when the destructor runs